### DISCUSSION TITLE
Ideas for Collaborative To-Do List Bot

### DISCUSSION THREAD

**Discussion by colintkn** [python-telegram-bot/python-telegram-bot#2751] [Q&A] [answered]
Hi all, 

Is it possible to create a shared to-do list
1. the to-do is shared in a group chat
2.  A member in the group that completes a task taps the check box and it marks off as done
3. The bot updates the to-do list 
4. Everyone in the group chat can see the live changes 
5. Supports undo (e.g. the person who marks something as done, can undo it) 
6. The To-do is the same everyday

![telegram-cloud-photo-size-5-6325694612156559140-y](https://user-images.githubusercontent.com/65516485/139091591-95711d6c-bd90-4ef3-b277-7a53a34760d7.jpg)

I've been looking at some telegram bot examples. So far, there is no option for a check button. 
![image](https://user-images.githubusercontent.com/65516485/139092267-bd0a6c44-8225-4b43-82d9-c095f322abfd.png)

Examples use in-line buttons. However this is not feasible in my case as the list is very long.

**Comment by zeroone2numeral2** [comment]
I can think to 4 possible solutions:

1. adding an emoji (or simply using the list counter at the left of each item) to hide a [deep-link](https://core.telegram.org/bots#deep-linking), that brings the user to the private chat with the bot. There, the user can decide what to do with the item after tapping on "START" (mark as completed, edit, remove, etc.)
2. another way to solve this that involves deep-linking, is adding a deeplink somewhere in the list message (eg. at the end) that says "mark an item as solved". If you tap on the link, it will bring to the bot's private chat and let you pick the item you want to mark as solved from a _normal_ keyboard. It can be scrolled, so a long list shouldn't be a problem
2. letting users reply to the list message with commands, eg. "`!1`" to mark the item n° 1 as solved/done. This is the most noisy solution as each action is visible to anyone in the group because it requires you to send a message
3. turning the inline keyboard to a paginated keyboard, with another row of buttons to move to the next/previous page. All the users will see (and use) the same keyboard though, so moving to another page will edit the keyboard for everyone

**Reply by colintkn** [reply]
Thank you @zeroone2numeral2

**Comment by Poolitzer** [comment, marked as answer]
In addition to the other comment, one could try and use sending a poll, though its very uncommon and will be a bit cumbersome to implement

**Reply by colintkn** [reply]
Yes I was thinking that a telegram poll is exactly the interface I'm looking for
![image](https://user-images.githubusercontent.com/65516485/139287628-dfe6936c-9d7f-4323-add7-109699ab177e.png)

This will be the most intuitive. 

But I'm not sure how exactly to implement it. 
- Such that you can 'undo' clicking a task 

I'm open to more suggestions!

**Reply by Poolitzer** [reply]
You can redo the vote. On Desktop, right click on the poll, retract vote. On the android app, tap on the poll, retract vote.

Its a non usual setup, but if its a close enough group and you can get everyone to understand the system its probably the most efficient in app one.

**Reply by colintkn** [reply]
Ok I'll begin working on this. Thanks @Poolitzer

**Reply by Poolitzer** [reply]
you got it :D

**Reply by colintkn** [reply]
I realised that telegram limits the number of poll options to 10. 
telegram.error.BadRequest: Poll can't have more than 10 options

Is there any workaround this, to increase the limits.

**Reply by Poolitzer** [reply]
No. Polls can only have 10 options.

**Comment by amine-beep** [comment]
https://drive.google.com/drive/folders/11v5238aj-ptHPC_FLnnynBSD0YLGqlr4?usp=sharing
